proxy/proxy_test.go

```diff
@@ -1385,6 +1431,55 @@ var _ = Describe("Proxy", func() {
 				Expect(string(b)).To(MatchRegexp("gorouter_time:0"))
 			})
 
+			Context("in websocket requests", func() {
+				It("shows slowness in response_time, not gorouter_time", func() {
+					ln := test_util.RegisterWSHandler(r, "slow-ws-test", func(conn *websocket.Conn) {
+						msgBuf := make([]byte, 100)
+						n, err := conn.Read(msgBuf)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(string(msgBuf[:n])).To(Equal("HELLO WEBSOCKET"))
+
+						time.Sleep(time.Second)
+
+						_, _ = conn.Write([]byte("WEBSOCKET OK"))
+						conn.Close()
+					})
+					defer ln.Close()
+
+					conn := dialProxy(proxyServer)
+					wsConn, err := wsClient(conn, "ws://slow-ws-test")
+					Expect(err).NotTo(HaveOccurred())
+
+					_, err = wsConn.Write([]byte("HELLO WEBSOCKET"))
+					Expect(err).NotTo(HaveOccurred())
+
+					msgBuf := make([]byte, 100)
+					n, err := wsConn.Read(msgBuf)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(string(msgBuf[:n])).To(Equal("WEBSOCKET OK"))
+
+					Eventually(func() (int64, error) {
+						fi, err := f.Stat()
+						if err != nil {
+							return 0, err
+						}
+						return fi.Size(), nil
+					}).ShouldNot(BeZero())
+
+					//make sure the record includes all the data
+					//since the building of the log record happens throughout the life of the request
+					logBytes, err := os.ReadFile(f.Name())
+					Expect(err).NotTo(HaveOccurred())
+					logStr := string(logBytes)
+					Expect(strings.HasPrefix(logStr, "slow-ws-test - [")).To(BeTrue())
+					Expect(logStr).To(ContainSubstring(`"GET / HTTP/1.1" 101`))
+					Expect(logStr).To(ContainSubstring(`x_forwarded_for:"127.0.0.1" x_forwarded_proto:"http" vcap_request_id:`))
+					Expect(logStr).To(MatchRegexp(`response_time:1`))
+					Expect(logStr).To(MatchRegexp(`gorouter_time:0`))
```

Comment:
Perhaps instead of checking for `1` and `0`, you can get the values and make sure that the response_time > gorouter_time.